junit-platform-launcher/src/main/java/org/junit/platform/launcher/tagexpression/TagExpression.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.tagexpression;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import org.apiguardian.api.API;
+
+/**
+ * Factory method to parse an {@link Expression Expression}
+ * from a <em>tag expression string</em>.
+ *
+ * @since 1.1
+ */
+@API(status = INTERNAL, since = "1.1")
+public class TagExpression {
+
+	///CLOVER:OFF
+	private TagExpression() {
+		/* no-op */
+	}
+	///CLOVER:ON
+
+	public static ParseResult parseFrom(String infixTagExpression) {
+		return new Parser().parse(infixTagExpression);
+	}
```

Comment:
I think we should merge `Expression` and `TagExpression` (I prefer the latter name). Moreover, I think `parseFrom()` should return a `TagExpression`. Then, we can make `ParseResult` package-private.

Reply:
If `parseFrom()` returns a TagExpression/Expression how do you then communicate a parse error? In this case I do not like to throw an Exception, because it is not an exceptional thing for a parser to get invalid input. Thats why I prefer to keep `ParseResult` as return type of `parseFrom`.
This makes the method easier to re-use for example in the IDEA run dialogue if they start supporting tag expressions.

How about this:
1. move `parseFrom()` into Expression
1. delete `TagExpression`
1. rename `Expression` -> `TagExpression`

Reply:
You're right, I forgot about that use case. This plan sounds good. 👍

Reply:
done